src/recipes/custom-icons.md

```diff
@@ -0,0 +1,50 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Custom Icons
+
+By default, all of your custom types will use the `Node` icon. While this can be serviceable, you will likely want to add custom
+icons, especially if you plan to distribute this to others!
+
+All of the icons must be registered by class name in your `.gdextension` file.
+
+```toml
+[icons]
+
+MyClass = "res://addons/your_extension/file_name.svg"
+```
+
+```admonish hint
+Note that the path is based off of the `res://` path, like all other resources for Godot.
+This means that you will need to have a standard 
+installation path for your tool. It is recommended to make this in the "addons" folder because 
+this will clearly signify to users that it is a 
+directory of third party resources and code and that they likely shouldn't mess 
+with it unless they are certain of what they are doing.
+```
```

Comment:
Something is off with the line lengths here, maybe you accidentally broke them too early?

Reply:
Also, is `addons/` actually an established convention? If yes, is there some documentation for it?

Reply:
I'll have to check those lines in my editor

From the basis that GDScript plugins must be installed into the `addons/` folder, it seems reasonable (to me) that we suggest `addons/` as the folder for GDExtension by default, especially if writing using EditorPlugin to do typical plugin modifications to the editor interface and such. It also cleans things up for redistribution because the end user doesn't necessarily need to know that the plugin is written in rust rather than gdscript in order to install correctly. Though if you want this to be changed to be less opinionated I can make that change

Reply:
Makes sense in that case!

Could you insert a link to the docs where this `addons` for GDScript is explained?

Reply:
I just pushed a commit with a link to the "Installing Plugins" page of the docs, which includes a callout detailing how to tell if something is a plugin. I'd argue that section should be updated to include `.gdextension` files, but one step at a time